### ISSUE TITLE
`ghc-9.6.4-r1`: `USE=numa` broken

### ISSUE THREAD

**Issue by 0x6d6e647a** [gentoo-haskell/gentoo-haskell#1530]
When upgrading to GHC v9.6.4-r1, I get the following error during the source preparation build step:
`find: ‘/var/tmp/portage/dev-lang/ghc-9.6.4-r1/work/usr/bin’: No such file or directory`

[ghc-9.6.4-r1_build.log.txt](https://github.com/gentoo-haskell/gentoo-haskell/files/14784466/ghc-9.6.4-r1_build.log.txt)
[emerge_info.txt](https://github.com/gentoo-haskell/gentoo-haskell/files/14784467/emerge_info.txt)

**Comment by hololeap** [comment]
This may be related to #1527. It seems like the `llvm` USE flag is broken on >=ghc-9.4 when you're on glibc.

**Comment by 0x6d6e647a** [comment]
I added `>=dev-lang/ghc-9.4.8::haskell -llvm` to my `package.use` after #1527. I double checked and the `llvm` use flag is not set.

I check the ebuild differences between v9.4.8 (working with `llvm` USE flag disabled) and v9.6.4-r1 and I don't see anything significant that would explain why the WORKDIR isn't being setup correctly.

**Comment by hololeap** [comment]
The main reason I assumed it was `USE=llvm` was because that's the only difference I saw between your setup and mine. I have `dev-lang/ghc-9.6.4-r1:0/9.6.4::haskell` installed in a sandbox with `USE="gmp profile"`. Looking again, it could also be `USE=numa`.

**Comment by 0x6d6e647a** [comment]
You're right! I removed the `numa` USE flag and it built and installed no problem! Feel free to edit the issue title to reflect this. Thanks for all the help, I really appreciate it!

**Comment by hololeap** [comment]
I can't help but think that this is a duplicate of #1527.

1. I cannot reproduce with `ghc-9.6.4-r1` and `USE="gmp numa"` (exclusive).

2. In your log file, you have this:

   ```
   find: ‘/var/tmp/portage/dev-lang/ghc-9.6.4-r1/work/usr/bin’: No such file or directory
   sed: no input files
   ```

   The only place in the ebuild that could trigger this error is in the `llvmize` function:

   https://github.com/gentoo-haskell/gentoo-haskell/blob/12378d5a2c2d67fd06c516ba2b4c6771ed9b0531/dev-lang/ghc/ghc-9.6.4-r1.ebuild#L344-L348

   In fact, it doesn't look like the `numa` USE flag does anything except adjust the arguments to `configure` and pull in `sys-process/numactl` as a dependency.

**Comment by hololeap** [comment]
```
 * USE:        abi_x86_64 amd64 elibc_glibc gmp kernel_linux numa
```

```
checking for numa.h... yes
checking for numaif.h... yes
checking for numa_available in -lnuma... yes
```

```
>>> dev-lang/ghc-9.6.4-r1 merged.
```

---

Closing as duplicate of #1527

**Comment by 0x6d6e647a** [comment]
@hololeap I've confirmed your results. Got a clean build with the `numa` use flag as long as `llvm` use flag is not set. I apologize if I wasted your time I this. Thanks again!
